lib/internal/process/report.js

```diff
@@ -98,18 +34,107 @@ const report = {
       throw new ERR_INVALID_ARG_TYPE('err', 'Object', err);
 
     return nr.getReport(err.stack);
+  },
+  get directory() {
+    emitExperimentalWarning('report');
+    return nr.getDirectory();
+  },
+  set directory(dir) {
+    emitExperimentalWarning('report');
```

Comment:
maybe emit the warning on access of the parent report property instead? Would avoid some duplication.